### PR TITLE
feat(spanner): add samples for directed reads

### PR DESCRIPTION
**Pull request by harshachinta**
This PR adds samples for directed reads.

### CODE REVIEW

File: spanner/spanner_snippets/spanner/spanner_directed_read_options.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	"fmt"
+	"io"
+
+	"cloud.google.com/go/spanner"
+	sppb "cloud.google.com/go/spanner/apiv1/spannerpb"
+	"google.golang.org/api/iterator"
+)
+
+// [START spanner_directed_read]
+func directedReadOptions(w io.Writer, db string) error {
```
Comment (asthamohta) [inline spanner/spanner_snippets/spanner/spanner_directed_read_options.go:28]: The sample in the spec for this feature has some comments to explain the usage of this feature. Can you add those here?

Comment (harshachinta) [reply spanner/spanner_snippets/spanner/spanner_directed_read_options.go:48]: Done

File: spanner/spanner_snippets/spanner/spanner_directed_read_options.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	available location will be used to process the
+	request.
+	* `location:us-east1 type:READ_ONLY` --> The "READ_ONLY" type replica(s)
+	in location "us-east1" will be used to process the request.
+		IncludeReplicas also contains an option for AutoFailover which when set
+	Spanner will not route requests to a replica outside the
+	IncludeReplicas list when all the specified replicas are unavailable
+	or unhealthy. The default value is `false`
+*/
+// [START spanner_directed_read]
```
Comment (telpirion) [inline spanner/spanner_snippets/spanner/spanner_directed_read_options.go:47]: issue: Move region tag to above `import` statement.

File: spanner/spanner_snippets/spanner/spanner_directed_read_options.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	"fmt"
+	"io"
+
+	"cloud.google.com/go/spanner"
+	sppb "cloud.google.com/go/spanner/apiv1/spannerpb"
+	"google.golang.org/api/iterator"
+)
+
+/*
+	Shows how to run a query with directed read options.
```
Comment (telpirion) [inline spanner/spanner_snippets/spanner/spanner_directed_read_options.go:28]: issue: please reformat this comment. If the following information is available on a documentation page, link to that page here.

See these resources:
https://googlecloudplatform.github.io/samples-style-guide/#description
https://go.dev/doc/effective_go#commentary

File: spanner/spanner_snippets/spanner/spanner_directed_read_options.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+					{
+						Type: sppb.DirectedReadOptions_ReplicaSelection_READ_ONLY,
+					},
+				},
+				AutoFailover: true,
+			},
+		},
+	}
+
+	stmt := spanner.Statement{SQL: `SELECT SingerId, AlbumId, AlbumTitle FROM Albums`}
```
Comment (telpirion) [inline spanner/spanner_snippets/spanner/spanner_directed_read_options.go:82]: nit: expand truncated variable names for sample readability.

**Review by telpirion — CHANGES_REQUESTED** [review]

File: spanner/spanner_snippets/spanner/spanner_directed_read_options.go (hunk 4 above)
Comment (harshachinta) [reply spanner/spanner_snippets/spanner/spanner_directed_read_options.go:82]: @telpirion 
I am following the convention used by other samples to maintain consistency across all samples.
Reference: https://github.com/GoogleCloudPlatform/golang-samples/blob/main/spanner/spanner_snippets/spanner/spanner_dml_batch_update.go#L36

File: spanner/spanner_snippets/spanner/spanner_directed_read_options.go (hunk 2 above)
Comment (harshachinta) [reply spanner/spanner_snippets/spanner/spanner_directed_read_options.go]: Thanks. Done

File: spanner/spanner_snippets/spanner/spanner_directed_read_options.go (hunk 3 above)
Comment (harshachinta) [reply spanner/spanner_snippets/spanner/spanner_directed_read_options.go]: Comment refactored.

**Review by asthamohta — APPROVED** [review]

File: spanner/spanner_snippets/spanner/spanner_directed_read_options.go (hunk 4 above)
Comment (telpirion) [reply spanner/spanner_snippets/spanner/spanner_directed_read_options.go:82]: As a maintainer for this repo, I'm asking you to make this change. Let's not allow the mistakes of the past to haunt our future.

**Review by telpirion — CHANGES_REQUESTED** [review]

**Comment by telpirion** [comment]
The last engagement from the author of this PR was > 3 months ago. Closing.

**Comment by harshachinta** [comment]
@telpirion 
Can we please keep this PR open? This is because we need it to track the Client Library readiness for GA and other stages. We will merge the PR before GA.

**Pull request closed without merging**